src/python/pants/python/python_setup.py

```diff
@@ -50,6 +50,24 @@ def register_options(cls, register):
                 "on the format of constraint files and how constraints are applied in Pex and Pip."
             ),
         )
+        register(
+            "--use-all-requirements",
+            advanced=True,
+            # TODO: Should this default to True? Seems like a better initial experience for most
+            #  users (although they still require at least a trivial lockfile to get any benefit).
+            default=False,
```

Comment:
I think no, in part for that reason of needing a lockfile configured. Instead, we can suggest it in the docs.